app/models/health/patient.rb

```diff
@@ -279,6 +275,59 @@ class Patient < Base
       )
     end
 
+    def self.cp_2_engagement(on) # rubocop:disable Naming/MethodParameterName
+      ssm_patient_id_scope = Health::SelfSufficiencyMatrixForm.distinct.
+        completed.
+        allowed_for_engagement.
+        where(completed_at: (..on.to_time)).
```

Comment:
Should all of these date fields be limited to time after 4/1/2023?

Reply:
I don't think so -- they are still valid if if they were completed before 4/1.